### ISSUE TITLE
Undefined behavior for bincount for empty array (migrated from Trac #1387)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2940]
Original ticket http://projects.scipy.org/numpy/ticket/1387
Reported 2010-02-02 by atmention:cournape, assigned to unknown.

E.g.:

```
import numpy as np
x = np.bincount(np.array([], dtype=int))
```

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-04-01

The desired behavior would be to just return the empty array (similar to histogram), or if bins are given to return an array of the right size with all zeros. 

Same issue for digitize(), it also gives a ValueError.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-03-04

Fixed in 40f0844.
